lib/datadog/core/telemetry/http/transport.rb

```diff
@@ -13,18 +14,47 @@ module Http
         # Class to send telemetry data to Telemetry API
         # Currently only supports the HTTP protocol.
         class Transport
+          def self.build_agent_transport
+            agent_settings = Configuration::AgentSettingsResolver.call(Datadog.configuration)
+
+            Transport.new(
+              host: agent_settings.hostname,
+              port: agent_settings.port,
+              path: Http::Ext::AGENT_ENDPOINT
+            )
+          end
```

Comment:
Minor: This is one of those "I know this was already like this but...".

So we really missed doing this at the time, but after a bunch of refactoring, we basically have the `build_telemetry` method in components receiving an `agent_settings`, and then not making it available to `Transport`, which ends up needing to recompute stuff _again_. (This will even cause some warnings to be printed more than once).

Aka -- can I convince you to pass in the `agent_settings` from the component here, and use it directly? ;)

Reply:
Yes, sure!